### PR TITLE
Access requests implement ResourceWithLabels.

### PR DESCRIPTION
**Pull request by mdwn**
Access requests now implement ResourceWithLabels and a few utility functions have been added. These are all for making access requests easier to work with for the new access request watcher introduced in
https://github.com/gravitational/teleport/pull/19626.

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

**Review by jakule — COMMENTED** [review]
@mdwn Is this everything that needs to be added to make it work? I don't see any changes in the usage or any tests that would make sure that everything is working fine.

**Comment by jakule** [comment]
> > @mdwn Is this everything that needs to be added to make it work? I don't see any changes in the usage or any tests that would make sure that everything is working fine.
> 
> Yeah, it's a been of a wonky thing because I'm kind of actively working on this in a proof-of-concept and then trying to backport some of the work like this as I go.
> 
> That being said, I wasn't sure whether to add unit tests for this or not since there aren't currently any, and I think your comment makes me realize we need them. I'll add them in and update this PR.
> 
> Edit: On second thought, this is difficult to write tests for in a vacuum -- most of the "tests" here are validated by golang itself w/r/t ensuring AccessRequest implements that required interfaces -- so long as `AsResources()` compiles properly, it more or less validates that we've got what's needed here.

I think that we should assert that `AccessRequest` implements `ResourceWithLabels`. Otherwise, I can just remove that change, and no one will notice. Adding a test can potentially prevent us from regression.

**Comment by mdwn** [comment]
> > > @mdwn Is this everything that needs to be added to make it work? I don't see any changes in the usage or any tests that would make sure that everything is working fine.
> > 
> > 
> > Yeah, it's a been of a wonky thing because I'm kind of actively working on this in a proof-of-concept and then trying to backport some of the work like this as I go.
> > That being said, I wasn't sure whether to add unit tests for this or not since there aren't currently any, and I think your comment makes me realize we need them. I'll add them in and update this PR.
> > Edit: On second thought, this is difficult to write tests for in a vacuum -- most of the "tests" here are validated by golang itself w/r/t ensuring AccessRequest implements that required interfaces -- so long as `AsResources()` compiles properly, it more or less validates that we've got what's needed here.
> 
> I think that we should assert that `AccessRequest` implements `ResourceWithLabels`. Otherwise, I can just remove that change, and no one will notice. Adding a test can potentially prevent us from regression.

Good point. This test has been added.

File: api/types/access_request_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2022 Gravitational, Inc.
```
Comment (jakule) [inline api/types/access_request_test.go:2]: nit:
```suggestion
Copyright 2023 Gravitational, Inc.
```

**Review by jakule — APPROVED** [review]

File: api/types/access_request_test.go (hunk 1 above)
Comment (mdwn) [reply api/types/access_request_test.go:2]: Guess I'm not used to it yet!

**Pull request merged**